process/logsevents/pendingBalances.go

```diff
@@ -21,13 +22,30 @@ func (pbp *pendingBalancesProc) addInfo(receiver string, token string, tokenNonc
 	hexEncodedNonce := converters.EncodeNonceToHex(tokenNonce)
 	key := fmt.Sprintf("%s-%s-%s-%s", pendingBalanceIdentifier, receiver, token, hexEncodedNonce)
 
-	pbp.pendingBalances[key] = &data.AccountInfo{
-		Address:         fmt.Sprintf("%s-%s", pendingBalanceIdentifier, receiver),
-		Balance:         value,
-		TokenName:       token,
-		TokenIdentifier: converters.ComputeTokenIdentifier(token, tokenNonce),
-		TokenNonce:      tokenNonce,
+	_, found := pbp.pendingBalances[key]
+	if !found {
+		pbp.pendingBalances[key] = &data.AccountInfo{
+			Address:         fmt.Sprintf("%s-%s", pendingBalanceIdentifier, receiver),
+			Balance:         value,
+			TokenName:       token,
+			TokenIdentifier: converters.ComputeTokenIdentifier(token, tokenNonce),
+			TokenNonce:      tokenNonce,
+		}
+		return
 	}
+
+	balanceBigMap, ok := big.NewInt(0).SetString(pbp.pendingBalances[key].Balance, 10)
+	if !ok {
+		pbp.pendingBalances[key].Balance = value
+	}
+
+	valueBig, ok := big.NewInt(0).SetString(value, 10)
+	if !ok {
+		return
```

Comment:
if you return here the `pbp.pendingBalances[key].Balance` will remain unchanged. Is this intended?

Reply:
yes. But this should never happen. (added just extra check )

Reply:
ok.